Assets/Scripts/EnemyController.cs

```diff
@@ -107,13 +107,24 @@ void FindNavTarget()
         //we set destination for target to run less than every frame, cause it's computationally heavy over longer distances
         if (refreshTargetTimer <= 0)
         {
-            Transform closestPlayer = players[0].transform;
-            //find player closest to enemy
-            for(int i = 0; i < players.Length; i++)
+            Transform closestPlayer = null; 
+            List<GameObject> alivePlayers = new List<GameObject>();
+            foreach (GameObject player in players)
             {
-                if(Vector3.Distance(players[i].transform.position, transform.position) < Vector3.Distance(closestPlayer.position, transform.position)) //We can add in DDA here by multiplying the distances based on the player with a multiplier
+                if (player.GetComponent<PlayerManager>().health > 0)
```

Comment:
```suggestion
List<GameObject> alivePlayers = players.FindAll(
               delegate(GameObject player){
                      return player.GetComponent<PlayerManager>().health > 0;
               }
)
```
What if you do this? More efficient and better practice. Also, we have some code with this syntax already

Reply:
I could only mark that line of if sentence to apply suggestion, so it's just to create alivePlayers list